### ISSUE TITLE
Decimal floating-point

### ISSUE THREAD

**Issue by stephentyrone** [apple/swift-numerics#7]
Implement IEEE 754 Decimal128 (and possibly Decimal64 as well, though it's less important). This has a dependency on [fixed-width integer types](https://github.com/apple/swift-numerics/issues/4) because we want to be able to leverage UInt128 arithmetic for the implementation.

**Comment by paulrs** [comment]
We would like to second the request for this feature. For financial applications where values will not exceed the limits of UInt128 and where precision in values is key, having a Decimal128 implementation (or Decimal64) would be very helpful.

**Comment by djehrlich** [comment]
I agree with @paulrs that Decimal128 would be extremely valuable for financial applications.

**Comment by jasonbobier** [comment]
Especially since Foundation's Decimal causes a fatal error in this simple case:

let d = Decimal(exactly: 0)

and Apple seems unwilling to fix it (I reported it January 17, 2021).

**Comment by xwu** [comment]
@jasonbobier The fatal error was simply that the API was unimplemented; it's been implemented in the open-source swift-corelibs-foundation for some years now and Apple's proprietary implementation will pick up the changes on some timeline of their choosing.

**Comment by jasonbobier** [comment]
@xwu *sigh* "Some timeline of their choosing" is accurate. I have bugs from 2003 still open with them.

**Comment by KTSnowy** [comment]
Is there any update on this?

**Comment by mgriebling** [comment]
I have an implementation of this including Decimal32, Decimal64, and Decimal128.  Decimal32 is mostly tested but the other implementation require testing.  This is a pure Swift implementation based on the Intel Decimal code.

How would I upload this code so others can contribute?
